### PR TITLE
Add on_dispatched event

### PR DESCRIPTION
**Pull request by PhrozenByte**
This event is fired when a notification was sent to the notifications server. This is less useful as class-level callback, but very useful with `Notification`, because it allows for a better separation of concerns (separates creating and sending notifications).

**ToDo:**
* [x] After merging #203 `Notification.on_dispatched` must be changed to:
    ```python
        on_dispatched: Callable[[], Any] | None = field(default=None, repr=False)
    ```
* [x] After merging #205 `DesktopNotifierBackend.on_dispatched` must be changed to the following (also update the caller accordingly):
    ```python
        def handle_dispatched(self, identifier: str) -> None:
            if identifier in self._notification_cache:
                notification = self._notification_cache[identifier]
                if notification.on_dispatched:
                    notification.on_dispatched()
                    return
            if self.on_dispatched:
                self.on_dispatched(identifier)
    ```
* [x] Add `on_dispatched` to `DesktopNotifierSync` after merging #207

### CODE REVIEW

**Comment by samschott** [comment]
Thanks, I can see this being useful! Could you check if similar events are available on macOS and Windows?

**Comment by PhrozenByte** [comment]
This is a Desktop-Notifier-only event, i.e. doesn't rely on the notifications server and thus works with any backend.

**Comment by samschott** [comment]
Makes sense! I confused this with an event for notification being delivered, which would indeed need some information from the platforms.

File: src/desktop_notifier/backends/dbus.py [hunk 1]
```diff
@@ -260,6 +260,7 @@ async def get_capabilities(self) -> frozenset[Capability]:
             Capability.TITLE,
             Capability.TIMEOUT,
             Capability.URGENCY,
+            Capability.ON_DISPATCHED,
```
Comment (samschott) [inline src/desktop_notifier/backends/dbus.py:263]: opt: I'm not sure if we need to add this as a capability if its not backend dependent at all but rather fired by DesktopNotifier itself. But then again, its consistent with other event handlers being listed here :)

**Review by samschott — APPROVED** [review]

File: src/desktop_notifier/backends/dbus.py (hunk 1 above)
Comment (PhrozenByte) [reply src/desktop_notifier/backends/dbus.py:263]: Exactly my thinking as well :+1:

**Pull request merged**